helmfile/charts/gatekeeper/mutations/templates/ndot-amount-dns-policy.yaml

```diff
@@ -0,0 +1,35 @@
+{{- if $.Values.ndots.enabled }}
+{{- $path:= "" }}
+{{- if $.Values.ndots.labelSelector.enabled }}
+  {{- $path = .Values.ndots.labelSelector.matchLabels }}
+{{- end }}
+{{- range $key, $val := $path | default .Values.ndots.clusterWide }}
+---
+apiVersion: mutations.gatekeeper.sh/v1
+kind: Assign
+metadata:
+  name: ndot-amount-dns-policy-{{ $val }}
+spec:
+  applyTo:
+    - groups: [""]
+      kinds: ["Pod"]
+      versions: ["v1"]
+  match:
+    scope: Namespaced
+    kinds:
+      - apiGroups: ["*"]
+        kinds: ["Pod"]
+    {{- if $.Values.ndots.labelSelector.enabled }}
+    labelSelector:
+      matchLabels:
+        {{ $key }}: {{ $val }}
+    {{- end }}
+  location: "spec.dnsConfig"
+  parameters:
+    assign:
+      value:
+        options:
+        - name: ndots
+          value: "{{ $.Values.ndots.ndotAmount }}"
+{{- end }}
+{{- end }}
```

Comment:
What you could do instead is 

```suggestion
{{- if $.Values.ndots.enabled }}
{{ define "ndots-mutation" }}
---
apiVersion: mutations.gatekeeper.sh/v1
kind: Assign
metadata:
  {{- if .mutationLabelValue }}
  name: ndot-amount-dns-policy-{{ .mutationLabelValue }}
  {{- else }}
  name: ndot-amount-dns-policy
  {{- end }}
spec:
  applyTo:
    - groups: [""]
      kinds: ["Pod"]
      versions: ["v1"]
  match:
    scope: Namespaced
    kinds:
      - apiGroups: ["*"]
        kinds: ["Pod"]
    {{- if .mutationLabelValue }}
    labelSelector:
      matchLabels:
        {{ .mutationLabelKey }}: {{ .mutationLabelValue }}
    {{- end }}
  location: "spec.dnsConfig"
  parameters:
    assign:
      value:
        options:
        - name: ndots
          value: "{{ .Values.ndots.ndotAmount }}"
{{- end }}

{{- if $.Values.ndots.labelSelector.enabled }}
{{- range $key, $val := .Values.ndots.labelSelector.matchLabels }}
{{- include "ndots-mutation" (mustMergeOverwrite (dict "mutationLabelKey" $key "mutationLabelValue" $val) $) }}
{{- end }}
{{- else }}
{{- include "ndots-mutation" $ }}
{{- end }}
{{- end }}
```

Then you don't have to have this magic variable that has to be a particular length for it to work. And we still only get one thing to manage. I might've missed something but I think this should work.

Reply:
Fixed thou i had to change where the define and if statement in the beginning was because define does not like being under a if statement.